runtime/client/impersonator.go

```diff
@@ -132,7 +132,11 @@ func (i *Impersonator) clientForServiceAccountOrDefault() (rc.Client, *polling.S
 	}
 	i.setImpersonationConfig(restConfig)
 
-	restMapper, err := apiutil.NewDynamicRESTMapper(restConfig)
+	httpClient, err := rest.HTTPClientFor(restConfig)
+	if err != nil {
+		return nil, nil, err
+	}
+	restMapper, err := apiutil.NewDynamicRESTMapper(restConfig, httpClient)
```

Comment:
Please create a function in runtime/client/client.go named `NewDynamicRESTMapper` that wraps this code. Then in all our controllers we can use this helper.